crates/nu-cli/src/commands/history/history_.rs

```diff
@@ -44,6 +44,10 @@ impl Command for History {
     ) -> Result<PipelineData, ShellError> {
         let head = call.head;
 
+        let Some(history) = engine_state.history_config() else {
+            return Ok(PipelineData::empty());
```

Comment:
Made it so that `PipelineData::Empty` is returned if history is disabled. Should this error instead?

Reply:
If you haven't passed --no-history and history is disabled, then it should show an error. Otherwise, it's working as expected, right?

Reply:
Makes sense!

Reply:
I think I meant to say that there are four scenarios here that might should be covered.
1. --no-history is passed, history is enabled (error state)
2. --no-history is passed, history is disabled (running as expected)
3. --no-history is not passed, history is enabled (running as expected)
4. --no-history is not passed, history is disabled (error state)

So, maybe the error state ones should show an error somewhere?

Reply:
Right now the `--no-history` flag has a one-to-one correspondence to `EngineState::history_enabled`.  So, states 1 and 4 shouldn't be possible. (Also, it looks like we do not store the parsed CLI args anywhere, so we cannot check for the error cases anyways.)